### PR TITLE
Add submitSurvey useCase

### LINKED ISSUE #2278: Create a use-case for survey
This should hook up the graphQL mutation to the database

### PR DESCRIPTION
**Pull request by petereast**
#### Background

**Reviewers** - I can't remember what we decided with regards to which graphql definition to use - so I've just used the input types described as part of #2277

#### Any relevant tickets

Closes #2278

#### How has this been tested?
**Reviewers** ensure that these test requirements are also reviewed.

Please indicate the need for this change to be tested. If not please justify, if so then at what levels. 

- [ ] Unit / Integration tests
- [ ] Browser tests
- [ ] End2end tests

#### UI Changes

- [ ] Has been reviewed against Designs
- [ ] Necessary updates to styleguide

### CODE REVIEW

**Comment by petereast** [comment]
(In theory, this should _fail_ api-tests if the feature flagging is working correctly)

**Comment by petereast** [comment]
I've tested that calling the resolver manually correctly inserts the response into the database.

File: packages/component-submission/server/use-cases/submitSurveyUseCase.test.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+      },
+      {
+        // There shouldn't be duplicate question answers, but if there is, what happens?
+        questionId: 'question3',
+        text: 'What?',
+        answer: 'answer3',
+      },
+    ]
+
+    const result = makeResponseObject(input)
```
Comment (erezmus) [inline packages/component-submission/server/use-cases/submitSurveyUseCase.test.js]: This is not really testing the use case object, only the makeResponseObject method which is an implementation detail.

The test should be testing the actual Use Case object. See getSubmissionUseCase.test.js file for example

**Review by erezmus — CHANGES_REQUESTED** [review]

**Review by diversemix — APPROVED** [review]

**Review by diversemix — APPROVED** [review]

File: packages/component-submission/server/use-cases/submitSurveyUseCase.test.js [hunk 2]
```diff
@@ -0,0 +1,108 @@
+const { useCase, makeResponseObject } = require('./submitSurveyUseCase')
+
+describe('submitSurveyUseCase', () => {
+  describe('useCase', () => {
+    it('does not supress errors', () => {
+      class MockSurveyResponseThrowsOnSave {
+        constructor(_) {
```
Comment (Cooryd) [inline packages/component-submission/server/use-cases/submitSurveyUseCase.test.js]: You can avoid needing empty constructors by using object literals. Best practice in jest is to wrap a function that returns one in a `jest.fn` 

```
const MockSurveyResponseEntity = jest.fn(() => ({
  save: jest.fn(() => throw new Error('error'))
}))
```

**Review by erezmus — APPROVED** [review]

File: packages/component-submission/server/resolvers/submitSurveyResponse.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+async function submitSurveyResponse(_, { data }, { user: userId }) {
+  const { surveyId, submissionId, answers } = data
+
+  try {
+    return await submitSurveyUseCase(
+      { SurveyResponse },
+      {
+        surveyId,
+        submissionId,
+        answers,
```
Comment (Cooryd) [inline packages/component-submission/server/resolvers/submitSurveyResponse.js:15]: Does this contain the question string to? if so we should name it `questionsAndAnswers` or something along those lines

File: packages/component-submission/server/use-cases/submitSurveyUseCase.test.js (hunk 2 above)
Comment (petereast) [reply packages/component-submission/server/use-cases/submitSurveyUseCase.test.js]: Fixed - I also discovered that you're meant to return async expects - who knew :man_shrugging:

File: packages/component-submission/server/resolvers/submitSurveyResponse.js (hunk 3 above)
Comment (petereast) [reply packages/component-submission/server/resolvers/submitSurveyResponse.js:15]: I think we resolved this face-to-face, it's defined in the GQL where the answer is defined as containing a question - let's improve this later on outside the scope of this PR.

**Review by will-byrne — APPROVED** [review]

**Pull request merged**